src/components/ProteinViewer/utils.ts

```diff
@@ -1,7 +1,60 @@
 import { createSelector } from 'reselect';
 import { toPlural } from 'utils/pages/toPlural';
 import { NOT_MEMBER_DBS } from 'menuConfig';
+import { getTrackColor, EntryColorMode } from 'utils/entry-color';
 
+const dbs4SingleDomain = [
+  'pfam',
+  'smart',
+  'pirsf',
+  'ncbifam',
+  'hamap',
+  'sfld',
+  'cdd',
+  'profile',
+];
+
+const selectRepresentativeDomains = (domains: Record<string, unknown>[]) => {
+  const flatDomains = [];
+  for (const domain of domains) {
+    const { accession, short_name, source_database, integrated } = domain;
+    for (const location of domain.entry_protein_locations as Array<ProtVistaLocation>) {
+      for (const fragment of location.fragments) {
+        const { start, end } = fragment;
+        flatDomains.push({
+          accession,
+          short_name,
+          source_database,
+          integrated,
+          start,
+          end,
+          color: getTrackColor({ source_database }, EntryColorMode.MEMBER_DB),
+          length: end - start,
+          keep: true,
+        });
+      }
+    }
+  }
+  // const sortedDomains = flatDomains.sort((a, b) => b.length - a.length);
```

Comment:
Mhh, I actually commented it because I was not completely sure the sorting was necessary. So I was testing without it, and it worked all right for all the examples Typhaine provided. But I'm still unsure if there are edge cases that I haven't tested. Maybe I should just leave the sorting, just to be sure.